cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -123,20 +122,32 @@ pub struct CatalogClient {
 }
 
 impl CatalogClient {
-    pub fn new(baseurl: &str) -> Self {
+    pub fn new(baseurl: &str, extra_headers: Option<BTreeMap<String, String>>) -> Self {
         // Remove the existing output file if it exists so we don't merge with
         // a previous `flox` invocation
         if let Ok(path_str) = std::env::var(FLOX_CATALOG_DUMP_DATA_VAR) {
             let path = Path::new(&path_str);
             let _ = std::fs::remove_file(path);
         }
 
+        // convert to HeaderMap
+        let mut header_map = HeaderMap::new();
+        if extra_headers.is_some() {
```

Comment:
nonblocking: typically you would do this:
```rust
if let Some(headers) = extra_headers {
    ...
}
```
instead of checking `is_some` and then unwrapping.

Reply:
Nice, thank you!